### PR TITLE
Support converting collection-initializers to collection-expressions in more cases.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Old system only supported converting collection in statements.  This was necessary for collection-initializers as we analyze the statements after the creation-expression to see what values are added.

However, for collection-expressions this isn't needed as we can get the values from the existing initializer.  e.g. `new List<int> { 1, 2, 3 }` can be converted to `[1, 2, 3]` where that shows up, even without follow-up statements.

### CODE REVIEW

File: src/Analyzers/Core/Analyzers/UseCollectionInitializer/AbstractObjectCreationExpressionAnalyzer.cs [hunk 1]
```diff
@@ -67,22 +65,35 @@ protected ImmutableArray<TMatch> AnalyzeWorker(CancellationToken cancellationTok
...
+            {
+                var result =
+                    TryInitializeVariableDeclarationCase(semanticModel, syntaxFacts, rootExpression, statement, cancellationToken) ??
+                    TryInitializeAssignmentCase(semanticModel, syntaxFacts, rootExpression, statement, cancellationToken);
+                if (result != null)
+                    return result;
+            }
+
+            // Even if the above cases didn't work, we always support converting a `new List<int>()` collection over to
+            // a collection expression.  We just won't analyzer later statement.
```
Comment (CyrusNajmabadi) [inline src/Analyzers/Core/Analyzers/UseCollectionInitializer/AbstractObjectCreationExpressionAnalyzer.cs:82]:
```suggestion
            // a collection expression.  We just won't analyze later statements.
```

**Comment by CyrusNajmabadi** [comment]
@akhera99 ptal

**Comment by CyrusNajmabadi** [comment]
@akhera99 @genlu @Cosifne ptal.

**Review by genlu — APPROVED** [review]
:shipit:

**Pull request merged**